crates/rooch-types/src/transaction/ethereum.rs

```diff
@@ -74,33 +95,40 @@ impl EthereumTransaction {
             <Secp256k1RecoverableSignature as ToFromBytes>::from_bytes(&rsv_signature)
                 .expect("Invalid signature");
 
+        Ok(recoverable_signature)
+    }
+
+    pub fn into_address(&self) -> Result<EthereumAddress, RoochError> {
+        // Prepare the signed message (RLP encoding of the transaction)
+        let message = self.tx_hash().to_fixed_bytes();
+        let recoverable_signature = self.into_signature()?;
         // Recover with Keccak256 hash to a public key
         let public_key = recoverable_signature
             .recover_with_hash::<Keccak256>(&message)
             .expect("Failed to recover public key");
+        // Get the address
+        let address = EthereumAddress::from(public_key);
+
+        Ok(address)
+    }
+}
 
-        // Combine the scheme, recoverable signature and public key as the following order to construct the final signature
-        // 1. The Rooch crypto scheme
-        // 2. The RSV signature
-        // 3. The public key
-        let mut scheme_rsv_signature_pubkey = Vec::new();
-        scheme_rsv_signature_pubkey.push(BuiltinScheme::EcdsaRecoverable.flag());
-        scheme_rsv_signature_pubkey.extend_from_slice(&rsv_signature);
-        scheme_rsv_signature_pubkey.extend_from_slice(public_key.as_bytes());
-
-        // Parse the "scheme_rsv_signature_pubkey" signature
-        // 99 length with 1 byte scheme following with 65 bytes recoverable signature and 33 bytes public key
-        let signature: Signature = <EcdsaRecoverableRoochSignature as ToFromBytes>::from_bytes(
-            &scheme_rsv_signature_pubkey,
-        )
-        .unwrap()
-        .into();
-
-        Ok(signature)
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct EthereumTransaction {
```

Comment:
Why we need this struct?

Reply:
I'm just copying the ones from the `RoochTransaction` in `rooch.rs` to facilitate passing `authenticator` as an arg to use in functions like `sign`, etc.

Reply:
The Ethereum original tx has the signature. We need to submit the original ethereum tx to the RPC.

Reply:
Done.